.github/workflows/multi_build.yml

```diff
@@ -6,23 +6,58 @@ on:
   pull_request:
 
 jobs:
+  setup-tools:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.21"
+          check-latest: true
+      - name: Cache Tools
+        id: tool-cache
+        uses: actions/cache@v3
+        with:
+          path: /home/runner/go/bin
+          key: tooling-${{ runner.os }}-${{ hashFiles('./internal/tools/go.sum') }}
+      - name: Install Tools
+        if: steps.tool-cache.outputs.cache-hit != 'true'
+        run: make install-tools
+
   build_linux:
     runs-on: ubuntu-latest-4-cores
+    needs:
+      - setup-tools
     steps:
       - name: Checkout Sources
         uses: actions/checkout@v4
       - name: Install Go
         uses: actions/setup-go@v5
         with:
-          go-version-file: go.mod
+          go-version: "1.21"
+          check-latest: true
+      - name: Cache Tools
+        id: tool-cache
+        uses: actions/cache@v3
+        with:
+          path: /home/runner/go/bin
+          key: tooling-${{ runner.os }}-${{ hashFiles('./Makefile') }}
```

Comment:
It looks like this is repeated quite a bit in CI. Can we make sure we are using the tools go.sum everywhere we are using the tools cache?

Reply:
yep, I should've looked for them the first time